### DISCUSSION TITLE
【更新日志】2024/03/17 PaoPaoDNS 改进SHUFFLE功能，增加trnc选项

### DISCUSSION THREAD

**Discussion by kkkgo** [kkkgo/PaoPaoDNS#109] [更新日志]
关于SHUFFLE功能的`lite`选项可参考上个版本更新日志：https://github.com/kkkgo/PaoPaoDNS/discussions/108  
## trnc选项：截断记录
改进SHUFFLE功能，增加trnc选项。该选项在lite选项的基础之上，如果返回的记录大于3个，则每次洗牌完成后仅在ttl有效期内输出3个随机记录。

举例：
`lite`模式下：
![image](https://github.com/kkkgo/PaoPaoDNS/assets/4530897/2654d6ab-4642-4627-b695-47acb63eedfe)
`trnc`模式下：
![image](https://github.com/kkkgo/PaoPaoDNS/assets/4530897/1c47f2c1-926d-481f-8d3b-b9f34635faa2)

该选项可以增强某些终端请求dns的兼容性，某些终端对于DNS请求没有正确处理`UDP payload size`，`trnc`选项可以每次仅随机提供3个记录，对多余记录进行truncate，尽量保证大小在默认值512内，可以增强终端处理的兼容性和避免使用tcp协议进行请求重试。一些公共DNS，如`114.114.114.114`也会进行类似的truncate处理。

**Comment by yyysuo** [comment]
重试会慢吗，dns只监听udp，确实会有些网站无法访问。

**Reply by kkkgo** [reply]
按现代系统来说，一般是UDP payload size是4096，大多数情况是足够的。DNS规范超过包大小就会要求使用TCP重新请求。

**Reply by yyysuo** [reply]
www.kaola.com/personal/my_order.html
这个结果就超了，不重传或者压缩，就访问不了。

**Reply by kkkgo** [reply]
你发的是网址不是域名，不知道你说的结果是什么。
